plugins/image2/plugin.js

```diff
@@ -942,7 +942,8 @@
 
 			// No center wrapper has been found.
 			else if ( name == 'figure' && el.hasClass( captionedClass ) ) {
-				image = el.getFirst( 'img' ) || el.getFirst( 'a' ).getFirst( 'img' );
+				var anchor = el.getFirst( 'a' );
+				image = el.getFirst( 'img' ) || ( anchor && anchor.getFirst( 'img' ) );
```

Comment:
I wonder if [`find`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_htmlParser_element.html#method-find) wouldn't be more readable here:

```javascript
image = el.find( function( child ) {
	return child.name === 'img' &&
		CKEDITOR.tools.array.indexOf( [ 'figure', 'a' ], child.parent.name ) !== -1;
}, true )[ 0 ];
```

The issue I have with [`getFirst`](https://ckeditor.com/docs/ckeditor4/latest/api/CKEDITOR_htmlParser_element.html#method-getFirst) is the fact that it works in totally counter-intuitive way, changing behaviour when parameter is passed (bold's mine):

> Gets this element's first child. If condition is given, this method returns the first child **which satisfies that condition**.

Maybe the version with `find` is more verbose, yet also more explicit and easier to understand – without the need to consult the docs.